pandas/tests/frame/methods/test_shift.py

```diff
@@ -656,3 +656,45 @@ def test_shift_axis1_many_periods(self):
 
         shifted2 = df.shift(-6, axis=1, fill_value=None)
         tm.assert_frame_equal(shifted2, expected)
+
+    def test_shift_with_iterable(self):
+        # GH#44424
+        data = {"a": [1, 2, 3], "b": [4, 5, 6]}
+        shifts = [0, 1, 2]
+
+        df = DataFrame(data)
+        shifted = df.shift(shifts)
+
+        expected = DataFrame(
+            {
+                "a_0": [1, 2, 3],
+                "b_0": [4, 5, 6],
+                "a_1": [np.NaN, 1.0, 2.0],
+                "b_1": [np.NaN, 4.0, 5.0],
+                "a_2": [np.NaN, np.NaN, 1.0],
+                "b_2": [np.NaN, np.NaN, 4.0],
+            }
+        )
+        tm.assert_frame_equal(expected, shifted)
+
+        # test pd.Series
+        s: Series = df["a"]
+        df_one_column: DataFrame = df[["a"]]
+        tm.assert_frame_equal(s.shift(shifts), df_one_column.shift(shifts))
```

Comment:
I think this should be moved to tests/series

Reply:
👍

Reply:
Hm - there is no `pandas/tests/series/methods/test_shift.py` ... It would feel odd to create it just for this purpose. And there are a lot of tests of series.shift here in `pandas/tests/frame/methods/test_shift.py`. Maybe it would be enough to just add an extra function for the series test here?

Reply:
Ah, thanks. Makes sense to add the test to this file and we can consider moving it to tests/series separately.